lightly/models/moco.py

```diff
@@ -58,87 +40,28 @@ class ResNetMoCo(nn.Module, _StateDictLoaderMixin):
     """
 
     def __init__(self,
-                 name: str ='resnet-18',
-                 width: float = 1.,
+                 backbone: nn.Module,
                  num_ftrs: int = 32,
                  out_dim: int = 128,
-                 num_splits: int = 8,
-                 m: float = 0.999):
-
-        self.num_ftrs = num_ftrs
-        self.out_dim = out_dim
-        self.num_splits = num_splits
-        self.m = m
+                 m: float = 0.999,
+                 batch_shuffle: bool = False):
 
-        super(ResNetMoCo, self).__init__()
-
-        self.features, self.projection_head = \
-            _get_features_and_projections(
-                ResNetGenerator(name=name, width=width, num_splits=num_splits),
-                self.num_ftrs,
-                self.out_dim,
-                num_splits
-            )
-
-        self.key_features, self.key_projection_head = \
-            _get_features_and_projections(
-                ResNetGenerator(name=name, width=width, num_splits=num_splits),
-                self.num_ftrs,
-                self.out_dim,
-                num_splits
-            )
-
-        # set key-encoder weights to query-encoder weights
-        for param_k in self.key_features.parameters():
-            param_k.requires_grad = False
-        for param_k in self.key_projection_head.parameters():
-            param_k.requires_grad = False
-        self._momentum_update_key_encoder(0.)
-
-    def load_from_state_dict(self,
-                             state_dict,
-                             strict: bool = True,
-                             apply_filter: bool = True):
-        """Initializes a ResNetMoCo and loads weights from a checkpoint.
+        super(MoCo, self).__init__()
 
-        Args:
-            state_dict:
-                State dictionary with layer weights.
-            strict:
-                Set to False when loading from a partial state_dict.
-            apply_filter:
-                If True, removes the `model.` prefix from keys in the state_dict.
+        self.backbone = backbone
+        self.projection_head = _get_moco_projection_head(num_ftrs, out_dim)
+        self.momentum_features = None
+        self.momentum_projection_head = None
 
-        """
-        self._custom_load_from_state_dict(state_dict, strict, apply_filter)
-        self._momentum_update_key_encoder(0.)
```

Comment:
I think it is pretty much impossible to eliminate magic numbers in ML engineering but at least using a named parameter when you pass a number (if not a constant) would be helpful re. readability (obviously there is a limit to this but in general it is a best practice)

Reply:
True, the part you commented on was actually removed from the code but it happens again further down...